src/main/java/org/veupathdb/service/eda/ds/plugin/pass/MapMarkersOverlayPlugin.java

```diff
@@ -14,94 +14,109 @@
   }
 
   @Override
-  protected List<StreamSpec> getRequestedStreams(PieplotSpec pluginSpec) {
+  protected List<StreamSpec> getRequestedStreams(MapMarkersOverlaySpec pluginSpec) {
     return ListBuilder.asList(
       new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getOutputEntityId())
         .addVar(pluginSpec.getXAxisVariable())
-        .addVars(pluginSpec.getFacetVariable())
+        .addVar(pluginSpec.getGeoAggregateVariable())
         .addVar(pluginSpec.getLatitudeVariable())
         .addVar(pluginSpec.getLongitudeVariable()));
   }
 
   @Override
   protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
-    PieplotSpec spec = getPluginSpec();
+    MapMarkersOverlaySpec spec = getPluginSpec();
     String showMissingness = spec.getShowMissingness() != null ? spec.getShowMissingness().getValue() : "noVariables";
     String deprecatedShowMissingness = showMissingness.equals("FALSE") ? "noVariables" : showMissingness.equals("TRUE") ? "strataVariables" : showMissingness;
     String valueSpec = singleQuote(spec.getValueSpec().getValue());
+    String xVarType = getVariableType(spec.getXAxisVariable());
 
     Map<String, VariableSpec> varMap = new HashMap<String, VariableSpec>();
     varMap.put("xAxisVariable", spec.getXAxisVariable());
-    varMap.put("facetVariable1", getVariableSpecFromList(spec.getFacetVariable(), 0));
-    varMap.put("facetVariable2", getVariableSpecFromList(spec.getFacetVariable(), 1));
+    varMap.put("geoAggregateVariable", spec.getGeoAggregateVariable());
     varMap.put("latitudeVariable", spec.getLatitudeVariable());
     varMap.put("longitudeVariable", spec.getLongitudeVariable());
       
     useRConnectionWithRemoteFiles(dataStreams, connection -> {
       connection.voidEval(getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, 
           spec.getXAxisVariable(),
-          getVariableSpecFromList(spec.getFacetVariable(), 0),
-          getVariableSpecFromList(spec.getFacetVariable(), 1),
+          spec.getGeoAggregateVariable(),
           spec.getLatitudeVariable(),
           spec.getLongitudeVariable()));
       connection.voidEval(getVoidEvalVarMetadataMap(DEFAULT_SINGLE_STREAM_NAME, varMap));
 
       String viewportRString = getViewportAsRString(spec.getViewport());
       connection.voidEval(viewportRString);
       
+      String binReportValue = "NULL";
+      if (xVarType.equals("STRING")) {
+        // maybe i should have a warning here if they pass a BinSpec?
+        connection.voidEval("binWidth <- NULL");
+      } else {
+        BinSpec binSpec = spec.getBinSpec();
+        validateBinSpec(binSpec, xVarType);
+        binReportValue = binSpec.getType().getValue() != null ? binSpec.getType().getValue() : "binWidth";
+        String binWidth = connection.eval("numBinsToBinWidth(" + DEFAULT_SINGLE_STREAM_NAME + "[[map$variableId[map$plotRef == 'xAxisVariable']]], 8)").toString();
```

Comment:
A lot of stuff to unpack here. By default we will calculate the binWidth in R with `numBinsToBinWidth` but I would need some help with the `DEFAULT_SINGLE_STREAM_NAME + "[[map$variableId[map$plotRef == 'xAxisVariable']]]` bit. I'm assuming a pass through the data is made?  Is that on the subset?

We should go ahead with default/hardcoded number of bins == 8, and let the user tweak via bin **width** controls, but I wonder in the future we should let the user choose the number of bins?

Reply:
DEFAULT_SINGLE_STREAM_NAME is a reference to the data table which has been read into the rserve process, and everything inside the square brackets is essentially asking for the column id which is associated w the xaxis variable. the data at this point does always reflect the subset, but once its read into the r process we dont really need to worry about 'passes' through a stream. it just sits in memory.